projects/core/services/breakpoint.service.ts

```diff
@@ -0,0 +1,53 @@
+import {Inject, Injectable} from '@angular/core';
+import {WINDOW} from '@ng-web-apis/common';
+import {fromEvent, merge, Observable} from 'rxjs';
+import {map, share, startWith} from 'rxjs/operators';
+
+import {TuiMedia} from '../interfaces';
+import {TUI_MEDIA} from '../tokens';
+
+/**
+ * Service to provide the current breakpoint based on Taiga UI's media queries
+ */
+@Injectable()
+export class TuiBreakpointService extends Observable<string | null> {
+    constructor(@Inject(TUI_MEDIA) media: TuiMedia, @Inject(WINDOW) windowRef: Window) {
+        const breakpoints = tuiGetBreakpoints(media);
+        const events$ = breakpoints.map(breakpoint =>
+            fromEvent<MediaQueryListEvent>(
+                windowRef.matchMedia(breakpoint.query),
+                `change`,
+            ),
+        );
+        const media$ = merge(...events$).pipe(
+            map(event => event.media),
+            map(media => breakpoints.find(breakpoint => breakpoint.query === media)!),
+            startWith(tuiCurrentBreakpoint(breakpoints, windowRef)),
+            map(media => media.name),
+            share(),
+        );
+
+        super(subscriber => media$.subscribe(subscriber));
+    }
+}
+
+export interface TuiBreakpoint {
+    name: string;
+    query: string;
+}
+
+export function tuiGetBreakpoints(media: TuiMedia): TuiBreakpoint[] {
+    return Object.entries(media).map(([name, value]) => ({
+        name,
+        query: `(max-width: ${value - 1}px)`,
```

Comment:
Why do we need -1? Wouldn't that introduce inconsistency with CSS queries?

Reply:
I thought we emit when viewport hits media breakpoint, ie. matchMedia('max-width:767px') will emit 'onchange' event when viewport is 768px. should I remove -1 ?

Reply:
@zarghamkhandev so what do you think about my last comment? Is -1 required? Maybe you can add demo page for this service with a side by side comparison with CSS, this way it would be easy to test and we probably need a demo page anyway, so that people know about this service existence :)

Reply:
I am still of the opinion that -1 is the way to go. We dispatch event when we hit the corresponding exact breakpoint, not 1px more. 

I will create demo page :)

Reply:
`Why do we need -1?`


Do we need to add a description?

```ts
/**
 * @note:
 * because ....
*/
query: `(max-width: ${value - 1}px)`,
```